character.py

```diff
@@ -0,0 +1,66 @@
+class Character:
+    def __init__(self, name, desc, inv, hp, power, dialogue, race, currency):
+        self.name = name
+        self.desc = desc
+        self.inv = inv
+        self.hp = hp
+        self.power = power
+        self.dialogue = dialogue
+        self.race = race
+        self.currency = currency
+
+    def interact(self):
+        section_content = (f'{self.name}: "{self.dialogue}"\n'
+                           f'Description: {self.desc}\n'
+                           f'Inventory: {[item.name for item in self.inv]}\n'
+                           f'Currency: {self.currency}\n'
+                           f'HP: {self.hp}\n'
+                           f'Power: {self.power}\n'
+                           f'Race: {self.race}')
+        print("*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*")
+        print(section_content) # The CIA glow in the dark
+        print("*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*")
+
+    def check_health(self, current_scene):
+        if self.hp <= 0:
+            print(f"{self.name} has died.")
+            # Assuming current_scene is defined somewhere, you can remove the character from the scene
+            current_scene.characters.remove(self)
+        else:
+            print(f"{self.name} has {self.hp} HP remaining.")
+
+    def equip_best_weapon(self):
```

Comment:
This is a creative piece of functionality! I like the idea to equip the best item, rather than a random or having the player choose.